src/main/java/com/team766/framework3/RunnableWithContext.java

```diff
@@ -1,23 +1,9 @@
 package com.team766.framework3;
 
-import java.util.Collections;
 import java.util.Set;
 
-@FunctionalInterface
 public interface RunnableWithContext {
     void run(Context context);
 
-    default <M extends Mechanism<?>> M reserve(M mechanism) {
-        throw new UnsupportedOperationException(
-                "Default implementation does not support reserving mechanisms!");
-    }
-
-    default void release(Mechanism<?> mechanism) {
-        throw new UnsupportedOperationException(
-                "Default implementation does not support reserving mechanisms!");
-    }
-
-    default Set<Mechanism<?>> reservations() {
```

Comment:
IMO, all `RunnableWithContext` implementations should have to consider which `Mechanism`s they reserve, and very few of them will reserve no mechanisms, because that sort of implies they don't have any effects. (It's possible that there's a Procedure that just publishes a Status or something, but I suspect that will be rare.)

If we want a one-liner declaration of runnables, we should have a helper for that:
```java
public class InlineProcedure extends Procedure {
    private Consumer<Context> runnable;
    private Set<Mechanism<?>> reservations;

    public InlineProcedure(Set<Mechanism<?>> reservations, Consumer<Context> runnable) {
        this.runnable = runnable;
        this.reservations = reservations;
    }

    @Override
    public void run(Context context) {
        runnable.accept(context);
    }

    @Override
    public Set<Mechanism<?>> reservations() {
        return reservations;
    }
}
```
```java
rules.add(Rule.create(gamepad.getButton(InputConstants.XBOX_RT)).
    withNewlyTriggeringRunnable(() -> new InlineProcedure(
        Set.of(shooter),
        (context) -> shooter.shoot(Shooter.SHOOTER_ASSIST_SPEED)));
```
which we could consider sugaring as
```java
rules.add(Rule.create(gamepad.getButton(InputConstants.XBOX_RT)).
    withNewlyTriggeringRunnable(
        Set.of(shooter),
        (context) -> shooter.shoot(Shooter.SHOOTER_ASSIST_SPEED));
```